### PR TITLE
Refactor/tracing path validator

### PR DESCRIPTION
**Pull request by romanatarango**
### Scope & Purpose

Added a traced version of `PathValidator` according to `Tracify.md` from the EE.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] Backport for 3.9: *(Please link PR)*
- [ ] Backport for 3.8: *(Please link PR)*
- [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [x] Docs PR: `Tracify.md` in `EE/documents/internal/`.
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Review by adityamukho — COMMENTED** [review]
Some potentially unused aliases have been defined. Otherwise LGTM.

File: arangod/Graph/Enumerators/OneSidedEnumerator.h [hunk 1]
```diff
@@ -58,6 +58,7 @@ class OneSidedEnumerator : public TraversalEnumerator {
   using Step = typename Configuration::Step;  // public due to tracer access
   using Provider = typename Configuration::Provider;
   using Store = typename Configuration::Store;
+  using Validator = typename Configuration::Validator;
```
[the remark this answers is not in the document]
Comment (romanatarango) [reply arangod/Graph/Enumerators/OneSidedEnumerator.h:61]: No.

File: arangod/Graph/PathManagement/PathValidatorTracer.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+template<class PathValidatorImplementation>
+class PathValidatorTracer {
+ public:
+  using VertexRef = arangodb::velocypack::HashedStringRef;
+  using Provider = typename PathValidatorImplementation::ProviderImpl;
+  using Step = typename Provider::Step;
+  using PathStore = typename PathValidatorImplementation::PathStoreImpl;
+  using VertexUniquenessImpl = VertexUniquenessLevel;
+  using EdgeUniquenessImpl = EdgeUniquenessLevel;
```
[the remark this answers is not in the document]
Comment (romanatarango) [reply arangod/Graph/PathManagement/PathValidatorTracer.h:52]: Same

File: arangod/Graph/PathManagement/PathValidator.h [hunk 3]
```diff
@@ -60,6 +60,11 @@ class PathValidator {
   using Step = typename Provider::Step;
 
  public:
+  using ProviderImpl = Provider;
+  using PathStoreImpl = PathStore;
+  using VertexUniquenessImpl = VertexUniquenessLevel;
+  using EdgeUniquenessImpl = EdgeUniquenessLevel;
```
[the remark this answers is not in the document]
Comment (romanatarango) [reply arangod/Graph/PathManagement/PathValidator.h:66]: Same

**Comment by hkernbach** [comment]
Unrelated CI failure.

**Pull request merged**